packages/nova-react/src/eventing/nova-eventing-provider.test.tsx

```diff
@@ -380,3 +390,138 @@ describe("useUnmountEventing", () => {
+      });
+    };
+
+    const onNonInterceptClick = (event: React.SyntheticEvent) => {
+      eventing.bubble({
+        reactEvent: event,
+        event: {
+          originator: "notToBeIntercepted",
+          type: "TypeNotToBeIntercepted",
+        },
+      });
+    };
+    return (
+      <>
+        Component with two events
+        <button onClick={onInterceptClick}>
+          {name}: Fire event to be intercepted
+        </button>
+        <button onClick={onNonInterceptClick}>
+          {name}: Fire event without intercept
+        </button>
+      </>
+    );
+  };
+
+  const InterceptorTestComponent: React.FC<{
+    interceptor?: (event: EventWrapper) => Promise<EventWrapper | undefined>;
+  }> = ({ interceptor = defaultInterceptor }) => (
+    <NovaEventingProvider
+      eventing={parentEventing}
+      reactEventMapper={realMapper}
+    >
+      <ComponentWithTwoEvents name="outside" />
+      <NovaEventingInterceptor interceptor={interceptor}>
+        <ComponentWithTwoEvents name="inside" />
+      </NovaEventingInterceptor>
+    </NovaEventingProvider>
+  );
+
+  it("intercepts the event and does not bubble it up", () => {
+    const { getByText } = render(<InterceptorTestComponent />);
+    const button = getByText("inside: Fire event to be intercepted");
+    button.click();
+    expect(callbackToBeCalledOnIntercept).toHaveBeenCalled();
+    expect(bubbleMock).not.toHaveBeenCalled();
+  });
+
+  it("bubbles the event when interceptor returns the event", async () => {
+    const { getByText } = render(<InterceptorTestComponent />);
+    const button = getByText("inside: Fire event without intercept");
+    button.click();
+    expect(callbackToBeCalledOnIntercept).not.toHaveBeenCalled();
+    await waitFor(() => expect(bubbleMock).toHaveBeenCalled());
+  });
+
+  it("doesn't catch events outside of interceptor context", async () => {
+    const { getByText } = render(<InterceptorTestComponent />);
+    const button = getByText("outside: Fire event to be intercepted");
+    button.click();
+    expect(callbackToBeCalledOnIntercept).not.toHaveBeenCalled();
+    await waitFor(() => expect(bubbleMock).toHaveBeenCalled());
+  });
+
+  it("throws an error when rendered outside of NovaEventingProvider", () => {
+    console.error = jest.fn();
+    expect.assertions(1);
+    const InterceptorTestComponent: React.FC = () => (
+      <NovaEventingInterceptor interceptor={defaultInterceptor}>
+        <ComponentWithTwoEvents name="inside" />
+      </NovaEventingInterceptor>
+    );
+
+    expect(() => render(<InterceptorTestComponent />)).toThrow(
+      "Nova Eventing provider must be initialized prior to creating NovaEventingInterceptor!",
+    );
+  });
+
+  it("intercepts the event and bubbles up the event when interceptor returns the event anyway", async () => {
+    const interceptor = (eventWrapper: EventWrapper) => {
+      if (eventWrapper.event.originator === "toBeIntercepted") {
```

Comment:
Can we add the expect.assertion checks here? I don't want something to break and mean they pass by timing out...

Reply:
Unfortunately, `waitFor` and `expect.asseritons` are not compatible. `waitFor` calls callback in timeout until it succeeds and because it has expect inside, the number of calls to expect is unstable inside test, depending on how fast async operations complete. We need to leave it as is and have `expect.assertions` only in tests where we don't call `waitFor`